src/sentry/utils/recurring.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import absolute_import
+
+import time
+import logging
+import threading
+
+from six import iteritems
+from random import random
+
+
+tasks = {}
+logger = logging.getLogger(__name__)
+
+
+def register_recurring(interval=None, jitter=0.1, services=['worker', 'web']):
+    """Registers a function to run in the background for the given services
+    at a specific interval.  The defaults can be overridden by the
+    `RECURRING_SERVICE_TASKS` config.
+    """
+    def decorator(f):
+        f.__recurring_config_defaults__ = {
+            'interval': interval,
+            'jitter': jitter,
+            'services': services,
+        }
+        return f
+    return decorator
+
+
+def load_config():
+    from django.conf import settings
+    for task in settings.RECURRING_SERVICE_TASKS:
+        module_name, func_name = task.rsplit('.', 1)
+        mod = __import__(module_name, None, None, [func_name])
+        callback = getattr(mod, func_name)
+        config = dict(callback.__recurring_config_defaults__)
+        config.update(settings.RECURRING_SERVICE_TASKS[task])
+        config['callback'] = callback
+
+        for service in config['services']:
+            task_id = '%s:%s(%s)' % (
+                module_name,
+                func_name,
+                service,
+            )
+            if config['interval'] is None:
+                raise RuntimeError('No interval set for %s' % task_id)
+            tasks.setdefault(service, {})[task_id] = config
+
+
+def run_recurring_for_service(service):
+    logger.info('Starting recurring tasks for %s', service)
+    load_config()
+    to_run = tasks.setdefault(service, {})
```

Comment:
Seems we should be able to return early here if there are no tasks to run for this service, right? And avoid spawning a thread always with nothing to do.